### PR TITLE
[GreenCity] Fix unexpected null values in isFavorite and isSubscribed fields in EvenDto class #6266

### LINKED ISSUE #6266: [GreenCity] Fix unexpected null values in isFavorite and isSubscribed fields in EvenDto class
Fix unexpected null values in isFavorite and isSubscribed fields in EvenDto class

### PR DESCRIPTION
**Pull request by sergoliarnik**
## Summary Of Issue :
Fix unexpected null values in isFavorite and isSubscribed fields in EvenDto class

**Issue Link:**
https://github.com/ita-social-projects/GreenCity/issues/6266

## Summary Of Changes :
- Extract private methods for building dtos;
- Refactor tests;
- Change type of isFavorite and isSubscribed fields from wrapper to primitiva, because null value is not possible.

### CODE REVIEW

File: dao/src/main/java/greencity/repository/EventRepo.java [hunk 1]
```diff
@@ -97,4 +98,20 @@ public interface EventRepo extends JpaRepository<Event, Long>, JpaSpecificationE
...
+    /**
+     * Get all user's favorite events by user id.
+     *
+     * @param userId {@link Long}.
+     */
+    @Query(value = "SELECT e FROM Event e LEFT JOIN e.followers AS f WHERE f.id = :userId")
+    Set<Event> findAllFavoritesByUser(Long userId);
+
+    /**
+     * Get all user's favorite events by user id.
```
Comment (RomaMocherniuk) [inline dao/src/main/java/greencity/repository/EventRepo.java]: why do we have the same Javadoc for `findAllFavoritesByUser` and `findAllSubscribedByUser`?

File: dao/src/main/java/greencity/repository/EventRepo.java [hunk 2]
```diff
@@ -97,4 +98,20 @@ public interface EventRepo extends JpaRepository<Event, Long>, JpaSpecificationE
...
     @Query(value = "SELECT e FROM Event e LEFT JOIN e.followers AS f WHERE f.id = :userId")
     Page<Event> findAllFavoritesByUser(Long userId, Pageable pageable);
+
+    /**
+     * Get all user's favorite events by user id.
+     *
+     * @param userId {@link Long}.
+     */
+    @Query(value = "SELECT e FROM Event e LEFT JOIN e.followers AS f WHERE f.id = :userId")
+    Set<Event> findAllFavoritesByUser(Long userId);
```
Comment (RomaMocherniuk) [inline dao/src/main/java/greencity/repository/EventRepo.java]: Should it be Pagable?
Users could follow many events

File: dao/src/main/java/greencity/repository/EventRepo.java [hunk 3]
```diff
@@ -97,4 +98,20 @@ public interface EventRepo extends JpaRepository<Event, Long>, JpaSpecificationE
...
+
+    /**
+     * Get all user's favorite events by user id.
+     *
+     * @param userId {@link Long}.
+     */
+    @Query(value = "SELECT e FROM Event e LEFT JOIN e.attenders AS f WHERE f.id = :userId")
+    Set<Event> findAllSubscribedByUser(Long userId);
```
Comment (RomaMocherniuk) [inline dao/src/main/java/greencity/repository/EventRepo.java]: same here

File: dao/src/main/java/greencity/repository/EventRepo.java (hunk 2 above)
Comment (RomaMocherniuk) [reply dao/src/main/java/greencity/repository/EventRepo.java]: We already have `Page<Event> findAllFavoritesByUser(Long userId, Pageable pageable)`
Why do we need this one?

File: service/src/main/java/greencity/service/EventServiceImpl.java [hunk 4]
```diff
@@ -122,23 +124,25 @@ public void delete(Long eventId, String email) {
     public EventDto getEvent(Long eventId, Principal principal) {
         Event event =
             eventRepo.findById(eventId).orElseThrow(() -> new NotFoundException(ErrorMessage.EVENT_NOT_FOUND));
-        EventDto eventDto = modelMapper.map(event, EventDto.class);
+        EventDto eventDto;
```
Comment (RomaMocherniuk) [inline service/src/main/java/greencity/service/EventServiceImpl.java]: we do not need it. We could use `return` [here](https://github.com/ita-social-projects/GreenCity/pull/6267/files#diff-eee18a7834701a9c0911786df843dbb6c25712682bc01c3c89e8d2b61f16e158R130) or [here](https://github.com/ita-social-projects/GreenCity/pull/6267/files#diff-eee18a7834701a9c0911786df843dbb6c25712682bc01c3c89e8d2b61f16e158R132)

File: service/src/main/java/greencity/service/EventServiceImpl.java [hunk 5]
```diff
@@ -122,23 +124,25 @@ public void delete(Long eventId, String email) {
...
+            eventDto = modelMapper.map(event, EventDto.class);
         }
         return eventDto;
     }
 
     @Override
     public PageableAdvancedDto<EventDto> getAll(Pageable page, Principal principal) {
         Page<Event> events = eventRepo.findAllByOrderByIdDesc(page);
-        PageableAdvancedDto<EventDto> eventDtos = buildPageableAdvancedDto(events);
+        PageableAdvancedDto<EventDto> eventDtos;
```
Comment (RomaMocherniuk) [inline service/src/main/java/greencity/service/EventServiceImpl.java]: same here

File: service/src/main/java/greencity/service/EventServiceImpl.java [hunk 6]
```diff
@@ -567,4 +531,56 @@ private ZonedDateTime findLastEventDateTime(Event event) {
...
+            eventsPage.isFirst(),
+            eventsPage.isLast());
+    }
+
+    private PageableAdvancedDto<EventDto> buildPageableAdvancedDtoForUser(Page<Event> events,
+        User user) {
+        PageableAdvancedDto<EventDto> eventDtos = buildPageableAdvancedDto(events);
+
+        setSubscribes(eventDtos, user);
+        setFollowers(eventDtos, user);
```
Comment (RomaMocherniuk) [inline service/src/main/java/greencity/service/EventServiceImpl.java]: can we move it to `buildPageableAdvancedDto`?

**Comment by RomaMocherniuk** [comment]
Please resolve conflict and check the Sonar report
There is 1 Code smell

File: service/src/main/java/greencity/service/EventServiceImpl.java [hunk 7]
```diff
@@ -360,7 +321,7 @@ public void removeFromFavorites(Long eventId, String email) {
     @Override
     public PageableAdvancedDto<EventDto> searchEventsBy(Pageable paging, String query) {
         Page<Event> page = eventRepo.searchEventsBy(paging, query);
-        return buildPageableAdvancedDto(page);
+        return buildPageableAdvancedDto(page, null);
```
Comment (RomaMocherniuk) [inline service/src/main/java/greencity/service/EventServiceImpl.java]: Can we overload `buildPageableAdvancedDto` method w/o a second param?

**Review by RomaMocherniuk — APPROVED** [review]

**Review by Yurii-Midianyi — APPROVED** [review]

**Review by HelenSotnik — APPROVED** [review]

**Review by Maksym-Lenets — APPROVED** [review]

**Pull request merged**